Source/DiabloUI/button.cpp

```diff
@@ -38,10 +33,13 @@ bool HandleMouseEventButton(const SDL_Event &event, UiButton *button)
 		return false;
 	switch (event.type) {
 	case SDL_MOUSEBUTTONUP:
-		button->m_action();
-		return true;
+		if (button->IsPressed()) {
+			button->Activate();
+			return true;
+		}
+		return false;

```

Comment:
This addresses a minor bug. If you click and hold the mouse away from a dialog button, then drag the mouse over the UI button and release the left mouse button previously this would activate the button with no visual feedback. This is unexpected given you can click on a button then drag away to abort the action.